database/import.go

```diff
@@ -212,7 +255,7 @@ func validatePrevHashColumn(db *sqlx.DB) error {
 	}
 
 	if count != 1 {
```

Comment:
You should actually adjust the validation because now we have more power to do it correctly.
IMHO what should we check is:
1. `count = importCount` (already implemented)
2. `maxHeight = importCount - 1` (already implemented)
3. heightUniqueness (already implemented)
4. all previous hashes have match in a hash column, excluding check for genesis parent hash **(not implemented)**
5. checkpoints hashes exists in db at expected height **(not implemented)**

Reply:
For hashes you can probably use something like
```
select h.hash
from headers h
left join headers ph on h.previous_block = ph.hash
where h.height != 0 and ph.hash is null
```

Reply:
checking checkpoints could be done this way:
```
with checkpoints as
         (select 11111 as height, '0000000069e244f73d78e8fd29ba2fd2ed618bd6fa2ee92559f542fdb26e7c1d' as hash union
          select 33333 as height, '000000002dd5588a74784eaa7ab0507a18ad16a236e7b1ce69f00d7ddfb5d0a6' as hash union
          select 74000 as height, '0000000000573993a3c9e41ce34471c079dcf5f52a0e824a81e7f953b8661a20' as hash union
          select 105000 as height, '00000000000291ce28027faea320c8d2b054b2e0fe44a773f3eefb151d6bdc97' as hash union
          select 134444 as height, '00000000000005b12ffd4cd315cd34ffd4a594f430ac814c91184a0d42d2b0fe' as hash union
          select 168000 as height, '000000000000099e61ea72015e79632f216fe6cb33d7899acb35b75c8303b763' as hash union
          select 193000 as height, '000000000000059f452a5f7340de6682a977387c17010ff6e6c3bd83ca8b1317' as hash union
          select 210000 as height, '000000000000048b95347e83192f69cf0366076336c639f9b7228e9ba171342e' as hash union
          select 216116 as height, '00000000000001b4f4b433e81ee46494af945cf96014816a4e2370f11b23df4e' as hash union
          select 225430 as height, '00000000000001c108384350f74090433e7fcf79a606b8e797f065b130575932' as hash union
          select 250000 as height, '000000000000003887df1f29024b06fc2200b55f8af8f35453d7be294df2d214' as hash union
          select 267300 as height, '000000000000000a83fbd660e918f218bf37edd92b748ad940483c7c116179ac' as hash union
          select 279000 as height, '0000000000000001ae8c72a0b0c301f67e3afca10e819efa9041e458e9bd7e40' as hash union
          select 300255 as height, '0000000000000000162804527c6e9b9f0563a280525f9d08c12041def0a0f3b2' as hash union
          select 319400 as height, '000000000000000021c6052e9becade189495d1c539aa37c58917305fd15f13b' as hash union
          select 343185 as height, '0000000000000000072b8bf361d01a6ba7d445dd024203fafc78768ed4368554' as hash union
          select 352940 as height, '000000000000000010755df42dba556bb72be6a32f3ce0b6941ce4430152c9ff' as hash union
          select 382320 as height, '00000000000000000a8dc6ed5b133d0eb2fd6af56203e4159789b092defd8ab2' as hash union
          select 400000 as height, '000000000000000004ec466ce4732fe6f1ed1cddc2ed4b328fff5224276e3f6f' as hash union
          select 430000 as height, '000000000000000001868b2bb3a285f3cc6b33ea234eb70facf4dcdf22186b87' as hash union
          select 470000 as height, '0000000000000000006c539c722e280a0769abd510af0073430159d71e6d7589' as hash union
          select 510000 as height, '00000000000000000367922b6457e21d591ef86b360d78a598b14c2f1f6b0e04' as hash union
          select 552979 as height, '0000000000000000015648768ac1b788a83187d706f858919fcc5c096b76fbf2' as hash union
          select 556767 as height, '000000000000000001d956714215d96ffc00e0afda4cd0a96c96f8d802b1662b' as hash)
select ch.hash
from checkpoints ch
join headers h on ch.hash = h.hash and ch.height = h.height
where h.hash is null
```

But actually those parts 
```
select xxx as height, 'yyy' as hash
```
should be build dynamically in code with use of Checkpoints list from: internal/chaincfg/params.go:253

Reply:
After some brainstorming and discussion actually point 4 & 5 can be replaced only by checking if the latest (highest) checkpoint exist in db, this should ensure that all previous blocks are correct.

Reply:
A newer block (height 826638) and some older ones have been added to the checkpoints in params.go. Now when the base is loaded, the hash of the last checkpoint will be compared with the hash of a block of the same height in the database

Reply:
removed